packages/react/src/events.tsx

```diff
@@ -0,0 +1,42 @@
+// This file in the future will be part of JSX support
+import crypto from 'node:crypto'
+
+import { AsyncHandlerEventDataState, HandlerEventDataState, TemplateEvent } from './template'
+
+type Handler = ((payload: any) => any) | ((payload: any) => Promise<any>)
+
+const handlers = new Map<string, Handler>()
+
+export function handler(func: Handler): AsyncHandlerEventDataState | HandlerEventDataState {
+  const key = crypto.randomUUID()
+  handlers.set(key, func)
+  switch (func.constructor.name) {
```

Comment:
this is kind of a whacky way to do it.. although I can see how it is convenient

one concern is that the name may change if the code gets compressed by a build step somewhere

Reply:
Yeah, or it gets transpiled to a non-async environment, in that case it's all functions (or sometimes even generators). 

I think once we have JSX support, we should have two ways:
- pure function is passed `prop={func}` - no response to the server
- wrapped with handler - `prop={handler(func)}` - response to the server

I don't think there's other reliable way.

For now, we need to use function in both cases as we don't have JSX sugar, so I guess calling `handler` or `asyncHandler` depending on your use case is going to be easiest.